salt/modules/boto3_route53.py

```diff
@@ -1000,3 +1003,448 @@ def change_resource_record_sets(HostedZoneId=None, Name=None,
                     (Name or HostedZoneId), six.text_type(e))
             raise e
     return False
+
+def get_health_check_ids_by_name(name, region=None, key=None, keyid=None, profile=None):
+    '''
+    Find Ids of all health checks with the given name tag.
+    Note that this can return multiple health checks, since the same name can be used
+    in multiple health checks.
+
+    name
+        The name associated with the health check.
+    region
+        Region to connect to.
+    key
+        Secret key to be used.
+    keyid
+        Access key to be used.
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    CLI Example:
+    .. code-block:: bash
+        salt myminion boto3_route53.get_health_check_ids_by_name ANAME  \
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    conn = _get_conn(region=region, key=key, keyid=keyid, profile=profile)
+    ids = [z['Id'] for z in _collect_results(conn.list_health_checks, 'HealthChecks', {})]
+    args = {'ResourceType': 'healthcheck', 'ResourceIds': ids}
+    return [z['ResourceId'] for z in _collect_results(conn.list_tags_for_resources, 'ResourceTagSets', args) if _has_name_tag(z['Tags'], name)]
+
+def get_health_check_id_by_name(name, region=None, key=None, keyid=None, profile=None):
+    '''
+    Find Id of a health check with the given name (as tag).
+
+    name
+        The name associated with the health check.
+    region
+        Region to connect to.
+    key
+        Secret key to be used.
+    keyid
+        Access key to be used.
+    profile
+        Dict, or pillar key pointing to a dict, containing AWS region/key/keyid.
+    CLI Example:
+    .. code-block:: bash
+        salt myminion boto3_route53.get_health_check_id_by_name ANAME \
+                profile='{"region": "us-east-1", "keyid": "A12345678AB", "key": "xblahblahblah"}'
+    '''
+    ids = get_health_check_ids_by_name(name, region=region, key=key, keyid=keyid, profile=profile)
+
+    if len(ids) > 1:
+        log.error(
+            'Request matched more than one HealthCheck (%s). Refine your '
+            'criteria and try again.', [z['Id'] for z in ret]
+        )
+
+    return ids[0] if len(ids) > 0 else None
+
+def list_tags_for_resource(Type, Id, region=None, key=None, keyid=None, profile=None):
```

Comment:
This was based on this other existing method on this same module:

```
def find_hosted_zone(Id=None, Name=None, PrivateZone=None,
                     region=None, key=None, keyid=None, profile=None):
```

However there 'Id' and 'Name' are optional arguments instead of positional ones. It looks like on boto3 modules, all parameters which have direct relation with AWS's XML API, are kept on PascalCase, so I followed this "convention".

Reply:
That seems reasonable, if mildly (personally) distasteful 😂